### PR TITLE
scheduler(engine): Support executor selectors

### PR DESCRIPTION
**Pull request by liuzix**
Issue Number: ref #6270

### What is changed and how it works?
- Support scheduling by selectors
- Refactored `engine/servermaster/schedule` to used a chained filter design.
- Changes to public APIs (Open API) and configurations **will be added in another PR**.

#### Tests 

 - Unit test
 - Integration test
 
##### Will it cause performance regression or break compatibility?
No.

##### Do you need to update user documentation, design documentation or monitoring documentation?
No.

### Release note 

```release-note
None.
```

### CODE REVIEW

File: engine/servermaster/scheduler/scheduler.go [hunk 1]
```diff
@@ -51,93 +50,35 @@ func (s *Scheduler) ScheduleTask(
...
-		executorID, hasConstraint, err := s.placementConstrainer.GetPlacementConstraint(ctx, resource)
+	request *schedModel.SchedulerRequest,
+) (candidates []model.ExecutorID, retErr error) {
+	infos := s.infoProvider.GetExecutorInfo()
+	for id := range infos {
+		candidates = append(candidates, id)
+	}
+	for _, filter := range s.filters {
+		var err error
+		candidates, err = filter.GetEligibleExecutors(ctx, request, candidates)
```
[the remark this answers is not in the document]
Comment (liuzix) [reply engine/servermaster/scheduler/scheduler.go:78]: Commented

**Review by amyangfei — APPROVED** [review]

File: engine/servermaster/executor_manager.go [hunk 2]
```diff
@@ -52,6 +50,10 @@ type ExecutorManager interface {
 	WatchExecutors(ctx context.Context) (
 		snap map[model.ExecutorID]string, stream *notifier.Receiver[model.ExecutorStatusChange], err error,
 	)
+
+	// GetExecutorInfo implements the interface scheduler.executorInfoProvider.
+	// It is called by the scheduler as the source of truth for executors.
+	GetExecutorInfo() map[model.ExecutorID]schedModel.ExecutorInfo
```
Comment (gozssky) [inline engine/servermaster/executor_manager.go:56]: `GetExecutorInfo` -> `GetExecutorInfos`? Since it returns a map.

**Review by gozssky — COMMENTED** [review]
Rest LGTM.

**Review by gozssky — APPROVED** [review]

**Pull request merged**